Statement.php

```diff
@@ -147,17 +156,21 @@ public function errorInfo() {
 
     /**
      * {@inheritdoc}
+     *
+     * @SuppressWarnings("PHPMD.StaticAccess")
      */
     public function execute($params = null) {
-        $rawRequest = $this->mysqlToRequest->transform($this->query, $this->params);
-        $request    = $this->authStrategy->transformRequest($rawRequest);
-        $restClient = $this->restClientFactory->createOne($request->getCurlOptions());
+        $transformedQuery = SqlQuery::quoteUrl(SqlQuery::setParams($this->query, $params !== null ? $params : $this->params));
+        $rawRequest       = $this->mysqlToRequest->transform($transformedQuery);
+        $request          = $this->authStrategy->transformRequest($rawRequest);
+
+        $response = $this->restClient->send($request);
 
-        $method     = strtolower($request->getMethod());
-        $response   = $method === HttpMethods::GET || $method === HttpMethods::DELETE ? $restClient->$method($request->getUrlAndQuery()) : $restClient->$method($request->getUrlAndQuery(), $request->getPayload());
-        $statusCode = $response->getStatusCode();
+        $this->result = Result::create($transformedQuery, $this->formatter->decode($response->getContent()));
+        $this->id     = !empty($this->result['id']) ? $this->result['id'] : null;
+        krsort($this->result);
 
-        return $statusCode === 200 || ($method === HttpMethods::DELETE && $statusCode === 204) ? $this->onSuccess($response, $method) : $this->onError($request, $response);
```

Comment:
ok, das is ne frechheit :D. Dann würde ich das nur explizit machen + en error des treibers schmeißen (von restclient will ich ja eigtl nix mitbekommen):

```php
public function execute($params = null) {
        $transformedQuery = SqlQuery::quoteUrl(SqlQuery::setParams($this->query, $params !== null ? $params : $this->params));
        $rawRequest       = $this->mysqlToRequest->transform($transformedQuery);
        $request          = $this->authStrategy->transformRequest($rawRequest);
        
        try {
            $response = $this->restClient->send($request);
            $result   = new Result($transformedQuery, $this->formatter->decode($response->getContent()));

            $this->result = $result->get();
            $this->id     = $result->id();

            return true;
        }
    } catch(e) {
         // as the error handling proposed by doctrine
         // does not work, we use the way of PDO_mysql
         // which just throws the possible errors
         throw DoctrineRestDriverErrorIwas.create(e);
    }
```